homotopy-web/src/model/proof/signature.rs

```diff
@@ -155,7 +155,11 @@ impl Signature {
                     .push_onto(*node, SignatureItem::Folder("New folder".to_owned(), true));
             }
             SignatureEdit::MoveBefore(from, to) => self.0.reparent_before(*from, *to),
-            SignatureEdit::MoveInto(from, to) => self.0.reparent_under(*from, *to),
+            SignatureEdit::MoveInto(from, to) => {
+                if from != to {
```

Comment:
I think both `MoveBefore` and `MoveInto` should check that `from` is not a transitive parent of `to`. Otherwise, if you move a folder into a sub-folder of itself we'll trip the same assertion etc.

Reply:
Quite right, good catch. I think this is now fixed.